In [ ]:
%pip install -U python-woc pandas matplotlib tqdm
# please clear the output of this cell in your notebook before checking it in

In [2]:
from woc.remote import WocMapsRemote
from tqdm import tqdm
import pandas as pd
# creates the client
woc = WocMapsRemote( base_url="https://worldofcode.org/api/")
# if you got an API key
# woc = WocMapsRemote( base_url="https://worldofcode.org/api/", api_key="woc-XXXXXX-YYYYYY" )

# Now for each of the ten projects you were assigned get commits, e.g.
projects = ["apache/ctakes","Bioconductor/Biobase","open-sdr/openwifi","HARPgroup/r-dh-ecohydro","r-quantities/units","typelevel/frameless","arogozhnikov/einops","kelu124/echomods","stan-dev/loo","issp-center-dev/HPhi"]
list_df_commits = []
for prj in projects:
  # Convert GitHub repository names to WoC V2412 project names
  prj = prj.lower().replace('/','_',2)
  commits = woc.get_values('p2c', prj)
  df = pd.DataFrame(commits, columns=["sha1"])
  df['project'] = prj
  list_df_commits.append(df)
df = pd.concat(list_df_commits)
#perhaps save the list (if no errors) so you do not need to retrieve them again
df.to_csv('df_commits.csv')
df.head(1)

,sha1,project
0,000271f3940e5c277416b7dd2ede108da8aac4a7,apache_ctakes


In [3]:
# If the number of commits its not very large, you can try to get them all at the same time,
# The max batch size is 10
import time
# let us first split df['sha1'] in chunks
chunks = [df['sha1'][x:x+10] for x in range(0, len(df), 10)]
commit_data = []

for chunk in tqdm(chunks): # iterate over the commits
  # res, err = woc.show_content_many('commit',chunk.to_list())

  # commit.tch returns the same data but faster
  res, err = woc.get_values_many('commit.tch',chunk.to_list())
  res = {k: v[0] for k, v in res.items()}  # this conversion is necessary because of the internal implementation

  # to walk around the rate limit
  time.sleep(1)

  if err: # check for errors
    print('Got Errors', err)

  for commit_sha, commit in res.items():
    # flatten commit objects
    commit_data.append({
          'commit': commit_sha,
          'tree': commit[0],
          'parent': list(commit[1]),
          'author': commit[2][0],
          'author_time': int(commit[2][1]),
          'author_tz': commit[2][2],
          'committer': commit[3][0],
          'committer_time': int(commit[3][1]),
          'committer_tz': commit[3][2],
          'message': commit[4],
    })

df_commit_data = pd.DataFrame(commit_data)
df_commit_data = df_commit_data.merge(df, left_on='commit', right_on='sha1')
df_commit_data.to_csv('df_commit_data.csv', index=False)
df_commit_data.head(2)

  2%|▊                                       | 36/1766 [00:37<29:51,  1.04s/it]

Got Errors {'1256cce344f11ce68de8ec1cd57effe6fbb709df': 'Key 1256cce344f11ce68de8ec1cd57effe6fbb709df not found in /da5_fast/All.sha1c/commit_18.tch'}


  3%|█▎                                      | 57/1766 [00:58<29:23,  1.03s/it]

Got Errors {'1d48864ea30a469e44196d1df4d92b6e7af301d5': 'Key 1d48864ea30a469e44196d1df4d92b6e7af301d5 not found in /da5_fast/All.sha1c/commit_29.tch'}


  3%|█▎                                      | 59/1766 [01:00<29:22,  1.03s/it]

Got Errors {'1ec85fb26e3a14a9c6c0b311b447d7b34ae91724': 'Key 1ec85fb26e3a14a9c6c0b311b447d7b34ae91724 not found in /da5_fast/All.sha1c/commit_30.tch'}


  4%|█▋                                      | 74/1766 [01:16<29:08,  1.03s/it]

Got Errors {'27827c06c16ce6c8f212c21db4e81328fdee166d': 'Key 27827c06c16ce6c8f212c21db4e81328fdee166d not found in /da5_fast/All.sha1c/commit_39.tch'}


  5%|██                                      | 93/1766 [01:36<28:46,  1.03s/it]

Got Errors {'324ebe83e39622f393ff5579dd0afc29bcd0ea17': 'Key 324ebe83e39622f393ff5579dd0afc29bcd0ea17 not found in /da5_fast/All.sha1c/commit_50.tch'}


  6%|██▏                                    | 100/1766 [01:43<28:36,  1.03s/it]

Got Errors {'360a46a6fea7089f43729081e59a4986c74a7922': 'Key 360a46a6fea7089f43729081e59a4986c74a7922 not found in /da5_fast/All.sha1c/commit_54.tch'}


  6%|██▍                                    | 111/1766 [01:54<28:27,  1.03s/it]

Got Errors {'3c1e327158c98d4aaf7e81ada9c17032dd0b050a': 'Key 3c1e327158c98d4aaf7e81ada9c17032dd0b050a not found in /da5_fast/All.sha1c/commit_60.tch'}


  7%|██▊                                    | 130/1766 [02:14<28:10,  1.03s/it]

Got Errors {'45b2c81b29f01c9d68fbc4b304b5640b09efd7aa': 'Key 45b2c81b29f01c9d68fbc4b304b5640b09efd7aa not found in /da5_fast/All.sha1c/commit_69.tch'}


  8%|██▉                                    | 135/1766 [02:19<28:10,  1.04s/it]

Got Errors {'4788292848ff000a341eb26d390e9048a5047333': 'Key 4788292848ff000a341eb26d390e9048a5047333 not found in /da5_fast/All.sha1c/commit_71.tch'}


  8%|███                                    | 136/1766 [02:20<28:10,  1.04s/it]

Got Errors {'485f5c34ed62f8d18699bf68e4b90a3b783b4fe9': 'Key 485f5c34ed62f8d18699bf68e4b90a3b783b4fe9 not found in /da5_fast/All.sha1c/commit_72.tch'}


  9%|███▎                                   | 151/1766 [02:36<27:51,  1.03s/it]

Got Errors {'500903e1cc07196c03f3a210c92ab46297bdffef': 'Key 500903e1cc07196c03f3a210c92ab46297bdffef not found in /da5_fast/All.sha1c/commit_80.tch'}


  9%|███▍                                   | 155/1766 [02:40<27:44,  1.03s/it]

Got Errors {'51c03d921ca7b27134b7ef38de1181a9485be4e1': 'Key 51c03d921ca7b27134b7ef38de1181a9485be4e1 not found in /da5_fast/All.sha1c/commit_81.tch'}


  9%|███▌                                   | 163/1766 [02:48<27:29,  1.03s/it]

Got Errors {'568cadb15e7f876fe7fb4415328608d09dc98751': 'Key 568cadb15e7f876fe7fb4415328608d09dc98751 not found in /da5_fast/All.sha1c/commit_86.tch'}


 11%|████▍                                  | 203/1766 [03:29<26:58,  1.04s/it]

Got Errors {'6b8ce0e3a4c6d3ef582710d548205b2d517d0af6': 'Key 6b8ce0e3a4c6d3ef582710d548205b2d517d0af6 not found in /da5_fast/All.sha1c/commit_107.tch'}


 12%|████▊                                  | 219/1766 [03:46<26:40,  1.03s/it]

Got Errors {'72b6520ffe1ca19bbe8202011c1ee1ac021e249c': 'Key 72b6520ffe1ca19bbe8202011c1ee1ac021e249c not found in /da5_fast/All.sha1c/commit_114.tch'}


 13%|█████                                  | 228/1766 [03:55<26:27,  1.03s/it]

Got Errors {'76dde2be9737ed5a6d712fb9370b2f35e1e7ffd0': 'Key 76dde2be9737ed5a6d712fb9370b2f35e1e7ffd0 not found in /da5_fast/All.sha1c/commit_118.tch'}


 14%|█████▎                                 | 243/1766 [04:11<26:14,  1.03s/it]

Got Errors {'8050a207b7e6a3066d0c67d337a404e88be55f02': 'Key 8050a207b7e6a3066d0c67d337a404e88be55f02 not found in /da5_fast/All.sha1c/commit_0.tch'}


 16%|██████▍                                | 289/1766 [04:58<25:19,  1.03s/it]

Got Errors {'97755f6356fe841ce3970c7508ec653d279aa8a2': 'Key 97755f6356fe841ce3970c7508ec653d279aa8a2 not found in /da5_fast/All.sha1c/commit_23.tch'}


 18%|██████▉                                | 316/1766 [05:26<25:02,  1.04s/it]

Got Errors {'a6cf733aa06503c40f247e90d61fe669c44fff01': 'Key a6cf733aa06503c40f247e90d61fe669c44fff01 not found in /da5_fast/All.sha1c/commit_38.tch'}


 18%|███████▏                               | 325/1766 [05:35<24:41,  1.03s/it]

Got Errors {'ab1fec5b4884dc920173438a67a2049193db2974': 'Key ab1fec5b4884dc920173438a67a2049193db2974 not found in /da5_fast/All.sha1c/commit_43.tch'}


 19%|███████▍                               | 336/1766 [05:47<24:30,  1.03s/it]

Got Errors {'afddf3a8c775d3097b803b354c9ab55e7a635ac2': 'Key afddf3a8c775d3097b803b354c9ab55e7a635ac2 not found in /da5_fast/All.sha1c/commit_47.tch'}


 19%|███████▌                               | 343/1766 [05:54<24:24,  1.03s/it]

Got Errors {'b372f8964cd0f2746fb227f680ea31146f171eee': 'Key b372f8964cd0f2746fb227f680ea31146f171eee not found in /da5_fast/All.sha1c/commit_51.tch'}


 21%|████████                               | 363/1766 [06:14<24:06,  1.03s/it]

Got Errors {'bc40f39526286322405810abb01bda62d8fc82bf': 'Key bc40f39526286322405810abb01bda62d8fc82bf not found in /da5_fast/All.sha1c/commit_60.tch'}


 21%|████████▏                              | 373/1766 [06:25<23:55,  1.03s/it]

Got Errors {'c205a4cc7343803ec8de754ed261810db5116170': 'Key c205a4cc7343803ec8de754ed261810db5116170 not found in /da5_fast/All.sha1c/commit_66.tch'}


 22%|████████▋                              | 391/1766 [06:43<23:46,  1.04s/it]

Got Errors {'ca7c7cbdfe68a4210026fb1b0ff48f3b826e8224': 'Key ca7c7cbdfe68a4210026fb1b0ff48f3b826e8224 not found in /da5_fast/All.sha1c/commit_74.tch'}


 23%|████████▊                              | 398/1766 [06:50<23:29,  1.03s/it]

Got Errors {'ce388eda3095a9c84f781faf9dbd41221de01e1f': 'Key ce388eda3095a9c84f781faf9dbd41221de01e1f not found in /da5_fast/All.sha1c/commit_78.tch'}


 24%|█████████▎                             | 419/1766 [07:12<23:05,  1.03s/it]

Got Errors {'d939676715db36b0de986367e6c832a81e5e8bce': 'Key d939676715db36b0de986367e6c832a81e5e8bce not found in /da5_fast/All.sha1c/commit_89.tch'}


 25%|█████████▉                             | 448/1766 [07:42<22:36,  1.03s/it]

Got Errors {'e8b7a272d5bdb376245e9f2a5af96173fcece9d1': 'Key e8b7a272d5bdb376245e9f2a5af96173fcece9d1 not found in /da5_fast/All.sha1c/commit_104.tch'}


 27%|██████████▎                            | 468/1766 [08:03<22:16,  1.03s/it]

Got Errors {'f26a62f3ae3021cb348c6d5a2b9e6d0e8ce9781c': 'Key f26a62f3ae3021cb348c6d5a2b9e6d0e8ce9781c not found in /da5_fast/All.sha1c/commit_114.tch'}


 27%|██████████▍                            | 475/1766 [08:10<22:12,  1.03s/it]

Got Errors {'f659132182856c087def1947a80829943c63d877': 'Key f659132182856c087def1947a80829943c63d877 not found in /da5_fast/All.sha1c/commit_118.tch'}


 28%|██████████▊                            | 488/1766 [08:23<21:54,  1.03s/it]

Got Errors {'fc527430858aa2fbb4ed334df26d10bd882a07f0': 'Key fc527430858aa2fbb4ed334df26d10bd882a07f0 not found in /da5_fast/All.sha1c/commit_124.tch'}


 28%|██████████▊                            | 490/1766 [08:25<21:53,  1.03s/it]

Got Errors {'fe04bf1ac3debf3fe50f1506da0ddf3998d80d20': 'Key fe04bf1ac3debf3fe50f1506da0ddf3998d80d20 not found in /da5_fast/All.sha1c/commit_126.tch'}


 46%|█████████████████▉                     | 811/1766 [13:59<16:38,  1.05s/it]

Got Errors {'882208a87b6f3095ed1074a23b86f415842f7a96': 'Key 882208a87b6f3095ed1074a23b86f415842f7a96 not found in /da5_fast/All.sha1c/commit_8.tch'}


 49%|███████████████████                    | 865/1766 [14:55<15:39,  1.04s/it]

Got Errors {'01b4d5dcda4c528cd4c2b2e568aaded80700faa1': 'Key 01b4d5dcda4c528cd4c2b2e568aaded80700faa1 not found in /da5_fast/All.sha1c/commit_1.tch'}


 49%|███████████████████▏                   | 871/1766 [15:01<15:30,  1.04s/it]

Got Errors {'06f48b2efc526be5cd91eafb844607644771e073': 'Key 06f48b2efc526be5cd91eafb844607644771e073 not found in /da5_fast/All.sha1c/commit_6.tch'}


 49%|███████████████████▎                   | 872/1766 [15:02<15:28,  1.04s/it]

Got Errors {'07d4248a054fa7b306877bdbdcb4a444d6676ead': 'Key 07d4248a054fa7b306877bdbdcb4a444d6676ead not found in /da5_fast/All.sha1c/commit_7.tch'}


 50%|███████████████████▍                   | 880/1766 [15:11<15:21,  1.04s/it]

Got Errors {'0e300f52fd675a1c99237f128593ff17e71e2008': 'Key 0e300f52fd675a1c99237f128593ff17e71e2008 not found in /da5_fast/All.sha1c/commit_14.tch'}


 50%|███████████████████▌                   | 883/1766 [15:14<15:17,  1.04s/it]

Got Errors {'1129f6e6a5939d28b0a54c2f4b5b7024659187e5': 'Key 1129f6e6a5939d28b0a54c2f4b5b7024659187e5 not found in /da5_fast/All.sha1c/commit_17.tch'}


 51%|███████████████████▉                   | 901/1766 [15:32<15:00,  1.04s/it]

Got Errors {'21ae93452563e30d8e7f6ba08998f7075b58d356': 'Key 21ae93452563e30d8e7f6ba08998f7075b58d356 not found in /da5_fast/All.sha1c/commit_33.tch'}


 53%|████████████████████▌                  | 933/1766 [16:06<14:32,  1.05s/it]

Got Errors {'3f77d8f9bd82dfb81f270fce8b50374883b43580': 'Key 3f77d8f9bd82dfb81f270fce8b50374883b43580 not found in /da5_fast/All.sha1c/commit_63.tch'}


 53%|████████████████████▊                  | 943/1766 [16:16<14:22,  1.05s/it]

Got Errors {'4a0a8ec47892e573b0601b05717b199af2917163': 'Key 4a0a8ec47892e573b0601b05717b199af2917163 not found in /da5_fast/All.sha1c/commit_74.tch'}


 57%|██████████████████████                 | 998/1766 [17:14<13:19,  1.04s/it]

Got Errors {'7bd8bd0bf1acbf73b4b422f363e80fb73e216076': 'Key 7bd8bd0bf1acbf73b4b422f363e80fb73e216076 not found in /da5_fast/All.sha1c/commit_123.tch'}


 57%|█████████████████████▌                | 1004/1766 [17:20<13:12,  1.04s/it]

Got Errors {'819019f2c44d7c6967b4e4f9d16871cb69902365': 'Key 819019f2c44d7c6967b4e4f9d16871cb69902365 not found in /da5_fast/All.sha1c/commit_1.tch', '81c74d8faaee6dc38137cb8e9744de288ecca00f': 'Key 81c74d8faaee6dc38137cb8e9744de288ecca00f not found in /da5_fast/All.sha1c/commit_1.tch'}


 57%|█████████████████████▋                | 1007/1766 [17:23<13:06,  1.04s/it]

Got Errors {'846309a349868480bb74f17c19cfde70e3f9b3a7': 'Key 846309a349868480bb74f17c19cfde70e3f9b3a7 not found in /da5_fast/All.sha1c/commit_4.tch'}


 57%|█████████████████████▋                | 1008/1766 [17:24<13:05,  1.04s/it]

Got Errors {'84895e8df6378a6f9afe416859be8785796552e2': 'Key 84895e8df6378a6f9afe416859be8785796552e2 not found in /da5_fast/All.sha1c/commit_4.tch'}


 58%|██████████████████████▏               | 1029/1766 [17:46<12:46,  1.04s/it]

Got Errors {'9c793fa516a3de9a7e3027a83efa494ef532e6ee': 'Key 9c793fa516a3de9a7e3027a83efa494ef532e6ee not found in /da5_fast/All.sha1c/commit_28.tch'}


 59%|██████████████████████▏               | 1034/1766 [17:51<12:42,  1.04s/it]

Got Errors {'a13d83ba8c4948ce6edf78dd424bb3e5ed535b05': 'Key a13d83ba8c4948ce6edf78dd424bb3e5ed535b05 not found in /da5_fast/All.sha1c/commit_33.tch'}


 59%|██████████████████████▍               | 1042/1766 [18:00<12:33,  1.04s/it]

Got Errors {'abdb82a29da5b23a425177766426efad6b9a0c0e': 'Key abdb82a29da5b23a425177766426efad6b9a0c0e not found in /da5_fast/All.sha1c/commit_43.tch'}


 59%|██████████████████████▍               | 1043/1766 [18:01<12:30,  1.04s/it]

Got Errors {'ad761f6ffd5d78d3abfe4d4fe08162f343e27cdf': 'Key ad761f6ffd5d78d3abfe4d4fe08162f343e27cdf not found in /da5_fast/All.sha1c/commit_45.tch'}


 60%|██████████████████████▋               | 1055/1766 [18:13<12:16,  1.04s/it]

Got Errors {'b870b236e3b6da917b9149d7d424612fb8160a3b': 'Key b870b236e3b6da917b9149d7d424612fb8160a3b not found in /da5_fast/All.sha1c/commit_56.tch'}


 60%|██████████████████████▊               | 1059/1766 [18:17<12:15,  1.04s/it]

Got Errors {'bce85cf3a50f5c1bffda4030f00c4d758890b977': 'Key bce85cf3a50f5c1bffda4030f00c4d758890b977 not found in /da5_fast/All.sha1c/commit_60.tch'}


 60%|██████████████████████▊               | 1061/1766 [18:19<12:12,  1.04s/it]

Got Errors {'bf8c0c75045d14b0aeb666fc0e356f1cc55d2755': 'Key bf8c0c75045d14b0aeb666fc0e356f1cc55d2755 not found in /da5_fast/All.sha1c/commit_63.tch'}


 60%|██████████████████████▉               | 1065/1766 [18:23<12:11,  1.04s/it]

Got Errors {'c411de3ce13d74a2c41f6b4b11cd2d290b227984': 'Key c411de3ce13d74a2c41f6b4b11cd2d290b227984 not found in /da5_fast/All.sha1c/commit_68.tch'}


 61%|███████████████████████               | 1070/1766 [18:29<12:02,  1.04s/it]

Got Errors {'c862ee1bf2f889cffe696bdff3871f2706e29892': 'Key c862ee1bf2f889cffe696bdff3871f2706e29892 not found in /da5_fast/All.sha1c/commit_72.tch'}


 61%|███████████████████████▎              | 1084/1766 [18:43<11:49,  1.04s/it]

Got Errors {'d605ab10d0017f51e10486e348f6dd3491ca172a': 'Key d605ab10d0017f51e10486e348f6dd3491ca172a not found in /da5_fast/All.sha1c/commit_86.tch'}


 62%|███████████████████████▋              | 1103/1766 [19:03<11:26,  1.04s/it]

Got Errors {'e82c5cf68092ecaa2fb08639e0ebed73995532ab': 'Key e82c5cf68092ecaa2fb08639e0ebed73995532ab not found in /da5_fast/All.sha1c/commit_104.tch'}


 64%|████████████████████████▎             | 1127/1766 [19:28<11:02,  1.04s/it]

Got Errors {'ff21a8a7fd6a7755f006012b927fc913d0630390': 'Key ff21a8a7fd6a7755f006012b927fc913d0630390 not found in /da5_fast/All.sha1c/commit_127.tch'}


 78%|█████████████████████████████▌        | 1374/1766 [23:44<06:46,  1.04s/it]

Got Errors {'1345397f77b2e7445f74846c35d9de23073f3a35': 'Key 1345397f77b2e7445f74846c35d9de23073f3a35 not found in /da5_fast/All.sha1c/commit_19.tch'}


 81%|██████████████████████████████▊       | 1434/1766 [24:47<05:48,  1.05s/it]

Got Errors {'73872d97de8d64a94f5b9ae6eecde37dc15de82f': 'Key 73872d97de8d64a94f5b9ae6eecde37dc15de82f not found in /da5_fast/All.sha1c/commit_115.tch'}


 82%|███████████████████████████████       | 1443/1766 [24:57<05:37,  1.04s/it]

Got Errors {'82d58b07a4478cf71f8cd68d97e32e2aa97da1aa': 'Key 82d58b07a4478cf71f8cd68d97e32e2aa97da1aa not found in /da5_fast/All.sha1c/commit_2.tch'}


 91%|██████████████████████████████████▍   | 1601/1766 [27:41<02:50,  1.04s/it]

Got Errors {'53e9a5f2bbd700cea27e9d735e4ebd7d7f55ca2b': 'Key 53e9a5f2bbd700cea27e9d735e4ebd7d7f55ca2b not found in /da5_fast/All.sha1c/commit_83.tch'}


100%|██████████████████████████████████████| 1766/1766 [30:31<00:00,  1.04s/it]


,commit,tree,parent,author,author_time,author_tz,committer,committer_time,committer_tz,message,sha1,project
0,000271f3940e5c277416b7dd2ede108da8aac4a7,070248827887375dd013167482260bd68d487e5a,[1ade4dd32196aca1f542b9aa3599cff52f71c5d2],Sean Finan <seanfinan@apache.org>,1484780774,+0000,Sean Finan <seanfinan@apache.org>,1484780774,+0000,"Write XMIs in tree, write using addLast\n\ngit...",000271f3940e5c277416b7dd2ede108da8aac4a7,apache_ctakes
1,0016a61c5ab40ab4ba90b2e38be6ffab17d3bae9,a8522236c71bb35ed9f364807855a5354a34611d,[ecf32c5e3dcdfa619cf57e0f38e56add540aede0],Sean Patrick Murphy <spmurphy50@apache.org>,1318604266,+0000,Sean Patrick Murphy <spmurphy50@apache.org>,1318604266,+0000,Fixes for version 1.2.0.\n\ngit-svn-id: https:...,0016a61c5ab40ab4ba90b2e38be6ffab17d3bae9,apache_ctakes


In [4]:
# what does the first record looks like?
# commit  parents
v = commit_data[0]
# commit tree, [parent{s}], [author, unixtime, tz ], [ committer, unixtime, tz ], Commit_message ]
print(v)

{'commit': '000271f3940e5c277416b7dd2ede108da8aac4a7', 'tree': '070248827887375dd013167482260bd68d487e5a', 'parent': ['1ade4dd32196aca1f542b9aa3599cff52f71c5d2'], 'author': 'Sean Finan <seanfinan@apache.org>', 'author_time': 1484780774, 'author_tz': '+0000', 'committer': 'Sean Finan <seanfinan@apache.org>', 'committer_time': 1484780774, 'committer_tz': '+0000', 'message': 'Write XMIs in tree, write using addLast\n\ngit-svn-id: https://svn.apache.org/repos/asf/ctakes/trunk@1779398 13f79535-47bb-0310-9956-ffa450edef68\n'}


In [5]:
# Now that the data is retrieved, save it and commit to your GH fork
# First, we need to flatten info in order to export as csv
# our dataframe will have columns project,'commit, author, time, message'
dfinf = pd.DataFrame(columns=['project', 'commit', 'author', 'time', 'message'])

commit_data = df_commit_data.to_dict("records")

for k in commit_data:
  row = pd.Series({'project': k['project'], 'commit': k['commit'], 'author': k['author'], 'time': k['author_time'], 'message':k['message']})
  dfinf = pd.concat([dfinf, row.to_frame().T ], ignore_index=True)


In [6]:
# check if it has the right content
dfinf.head(1)

,project,commit,author,time,message
0,apache_ctakes,000271f3940e5c277416b7dd2ede108da8aac4a7,Sean Finan <seanfinan@apache.org>,1484780774,"Write XMIs in tree, write using addLast\n\ngit..."


In [7]:
#mode a means append, so you have all your projects in the same file
yournetid='dbritto3'
dfinf.to_csv(yournetid+'_project_summary.csv', index=False,sep=';', mode='w', header=['project_wocid', 'commit_sha1', 'author', 'time', 'commit message'])

In [9]:
project_stats = dfinf.groupby("project").agg(ncommits=("commit", "nunique"),nauthors=("author", "nunique"),earliest=("time", "min"),latest=("time", "max")).reset_index()

project_stats["earliest"] = pd.to_datetime(project_stats["earliest"], unit="s", utc=True)
project_stats["latest"] = pd.to_datetime(project_stats["latest"], unit="s", utc=True)

display(project_stats)

,project,ncommits,nauthors,earliest,latest
0,apache_ctakes,4914,61,2009-03-08 14:52:58+00:00,2025-09-26 02:51:31+00:00
1,arogozhnikov_einops,1011,64,2018-09-22 00:45:08+00:00,2025-08-17 03:11:57+00:00
2,bioconductor_biobase,1027,57,2001-10-04 15:55:43+00:00,2026-01-16 15:44:14+00:00
3,harpgroup_r-dh-ecohydro,475,16,2017-05-30 20:25:36+00:00,2025-05-02 16:50:20+00:00
4,issp-center-dev_hphi,2428,47,2015-09-11 02:40:52+00:00,2025-10-25 06:47:04+00:00
5,kelu124_echomods,1331,10,2016-04-12 18:57:33+00:00,2025-05-23 23:20:25+00:00
6,open-sdr_openwifi,1043,45,2019-11-08 11:07:21+00:00,2025-10-28 18:25:23+00:00
7,r-quantities_units,1130,50,2016-06-07 07:40:34+00:00,2025-10-10 08:14:18+00:00
8,stan-dev_loo,1607,45,2015-06-13 07:12:22+00:00,2025-11-04 06:55:26+00:00
9,typelevel_frameless,2630,109,2015-03-28 06:03:10+00:00,2025-11-03 20:31:19+00:00


# Make sure you check in to your fork not just the notebook but also the csv files!!!

# Don't forget to add requested data from github and this notebook
### For each of the 10 projects go to their github repo and get the number of stars, number of forks, and the last commit date
### Report (in your notebook) the number of commits, the number of authors, and max and min time for each project based on WoC commits and also add the info you obtained from github

*WoC returned 62 "Key not found" errors. The WoC statistics below show the commits that were successfully collected.

### apache_ctakes

GitHub:
- Stars: 138
- Forks: 27
- Last commit date: 2026-08-12

World of Code:
- Number of commits: 4914
- Number of authors: 61
- Earliest commit time: 2009-03-08 14:52:58
- Latest commit time: 2025-09-26 02:51:31

### arogozhnikov_einops

GitHub:
- Stars: 9605
- Forks: 402
- Last commit date: 2026-08-26

World of Code:
- Number of commits: 1011
- Number of authors: 64
- Earliest commit time: 2018-09-22 00:45:08
- Latest commit time: 2025-08-17 03:11:57

### bioconductor_biobase

GitHub:
- Stars: 9
- Forks: 11
- Last commit date: 2026-07-28

World of Code:
- Number of commits: 1027
- Number of authors: 57
- Earliest commit time: 2001-10-04 15:55:43
- Latest commit time: 2026-01-16 15:44:14

### harpgroup_r-dh-ecohydro

GitHub:
- Stars: 0
- Forks: 1
- Last commit date: 2025-05-02

World of Code:
- Number of commits: 475
- Number of authors: 16
- Earliest commit time: 2017-05-30 20:25:36
- Latest commit time: 2025-05-02 16:50:20

### issp-center-dev_hphi

GitHub:
- Stars: 104
- Forks: 29
- Last commit date: 2026-07-15

World of Code:
- Number of commits: 2428
- Number of authors: 47
- Earliest commit time: 2015-09-11 02:40:52
- Latest commit time: 2025-10-25 06:47:04

### kelu124_echomods

GitHub:
- Stars: 448
- Forks: 121
- Last commit date: 2026-04-08

World of Code:
- Number of commits: 1331
- Number of authors: 10
- Earliest commit time: 2016-04-12 18:57:33
- Latest commit time: 2025-05-23 23:20:25

### open-sdr_openwifi

GitHub:
- Stars: 4822
- Forks: 822
- Last commit date: 2026-09-21

World of Code:
- Number of commits: 1043
- Number of authors: 45
- Earliest commit time: 2019-11-08 11:07:21
- Latest commit time: 2025-10-28 18:25:23

### r-quantities_units

GitHub:
- Stars: 203
- Forks: 35
- Last commit date: 2026-09-22

World of Code:
- Number of commits: 1130
- Number of authors: 50
- Earliest commit time: 2016-06-07 07:40:34
- Latest commit time: 2025-10-10 08:14:18

### stan-dev_loo

GitHub:
- Stars: 157
- Forks: 39
- Last commit date: 2026-09-15

World of Code:
- Number of commits: 1607
- Number of authors: 45
- Earliest commit time: 2015-06-13 07:12:22
- Latest commit time: 2025-11-04 06:55:26

### typelevel_frameless

GitHub:
- Stars: 895
- Forks: 135
- Last commit date: 2026-08-28

World of Code:
- Number of commits: 2630
- Number of authors: 109
- Earliest commit time: 2015-03-28 06:03:10
- Latest commit time: 2025-11-03 20:31:19
